src/components/tools/geometry-tool/use-label-segment-dialog.tsx

```diff
@@ -0,0 +1,107 @@
+import { ESegmentLabelOption } from "../../../models/tools/geometry/jxg-changes";
+import { getPolygonEdge } from "../../../models/tools/geometry/jxg-polygon";
+import { useCustomModal } from "../../../hooks/use-custom-modal";
+import "./label-segment-dialog.sass";
+
+interface LabelRadioButtonProps {
+  display: string;
+  label: string;
+  checkedLabel: string;
+  setLabelOption: React.Dispatch<React.SetStateAction<string>>;
+}
+const LabelRadioButton: React.FC<LabelRadioButtonProps> = ({display, label, checkedLabel, setLabelOption}) => {
+  return (
+    <label className="radio-button-label">
+      <input
+        className="radio-button"
+        type="radio"
+        id={label}
+        name="labelOption"
+        value={label}
+        checked={label === checkedLabel}
+        onChange={e => {
+          if (e.target.checked) {
+            setLabelOption(e.target.value);
+          }
+        }}
+      />
+      {display}
+    </label>
+  );
+};
+
+interface IContentProps {
+  labelOption: string;
+  setLabelOption: React.Dispatch<React.SetStateAction<string>>;
+}
+const Content: React.FC<IContentProps> = ({ labelOption, setLabelOption }) => {
+  return (
+    <fieldset className="radio-button-set">
+      <LabelRadioButton
+        display="None"
+        label={ESegmentLabelOption.kNone}
+        checkedLabel={labelOption}
+        setLabelOption={setLabelOption}
+      />
+      <LabelRadioButton
+        display="Label"
+        label={ESegmentLabelOption.kLabel}
+        checkedLabel={labelOption}
+        setLabelOption={setLabelOption}
+      />
+      <LabelRadioButton
+        display="Length"
+        label={ESegmentLabelOption.kLength}
+        checkedLabel={labelOption}
+        setLabelOption={setLabelOption}
+      />
+    </fieldset>
+  );
+};
+
+function getPolygonSegment(board: JXG.Board, polygon: JXG.Polygon, points: [JXG.Point, JXG.Point]) {
+  const pointIds = points.map(pt => pt.id);
+  return getPolygonEdge(board, polygon.id, pointIds);
+}
+
+interface IProps {
+  board: JXG.Board;
+  polygon: JXG.Polygon;
+  points: [JXG.Point, JXG.Point];
+  onAccept: (polygon: JXG.Polygon, points: [JXG.Point, JXG.Point], labelOption: ESegmentLabelOption) => void;
+  onClose: () => void;
+}
+export const useLabelSegmentDialog = ({ board, polygon, points, onAccept, onClose }: IProps) => {
+  // TODO: Do we need to find the segment on every render?
+  const segment = getPolygonSegment(board, polygon, points);
+  const [initialLabelOption] = useState(segment?.getAttribute("clientLabelOption") || "none");
+  const [labelOption, setLabelOption] = useState(initialLabelOption);
+
+  const handleClick = () => {
+    if (polygon && points && (initialLabelOption !== labelOption)) {
+      onAccept(polygon, points, labelOption);
+    } else {
+      onClose();
+    }
+  };
+
+  const [showModal, hideModal] = useCustomModal({
+    Icon: LineLabelSvg,
+    title: "Edit Segment Label",
```

Comment:
"Line Label" would be a better title.

Reply:
Should this be "Segment Label" since they aren't true lines?

Reply:
Technically, you are correct that "Segment" would be more accurate, but my main goal was consistency with the tooltip for the button that launches the dialog, which is currently "Line Label" and which is specified as such in the design spec (https://app.zeplin.io/project/5d62a554d64a9e02dcad80de/screen/5f4514c67282bc3f9013b84b), which you may not have access to yet. So for now I'd make it "Line Label" and we can raise the question of whether both the tooltip and the dialog title should be changed to "Segment Label" in slack at some point.

Reply:
Alright, switched it to "Line Label". It's not difficult to change later.